src/dvc_data/index/diff.py

```diff
@@ -167,19 +167,22 @@ def _diff(
     meta_cmp_key: Optional[Callable[[Optional["Meta"]], Any]] = None,
     shallow: Optional[bool] = False,
     callback: Callback = DEFAULT_CALLBACK,
+    start: Optional["DataIndexKey"] = None,
```

Comment:
Instead of setting default to None, you can set it to `()`.

```suggestion
    start: DataIndexKey = (),
```